system/src/system_cloud_internal.cpp

```diff
@@ -475,6 +478,24 @@ void SystemEvents(const char* name, const char* data)
                 System.reset();
         }
     }
+    if (!strncmp(name, KEY_RESTORE_EVENT, strlen(KEY_RESTORE_EVENT))) {
```

Comment:
The PSK and server address can be restored in RAM from system flash... so you can connect again. We decided not to automatically restore to DCT/DCD in case of conditions still present that may have caused the first corruption; perhaps they may further corrupt something else during the next flash write. We will track metrics on units reporting corruption, and in the future enable a way for users to send the restore event.